PHP/PHP Source.sublime-syntax

```diff
@@ -589,12 +592,61 @@ contexts:
         - include: identifier-constant-pop
     - include: use-statement-common
     - include: identifier-constant-pop
+  enum:
+    - match: '(?i)\b(enum)\s+({{identifier}})\s*'
+      captures:
+        1: keyword.declaration.enum.php
+        2: entity.name.enum.php
+      push: enum-definition
+
+  enum-definition:
+    - match: "(?=;)"
+      pop: 1
+    - match: "(?=:)"
+      push:
+        - match: ':'
+          scope: punctuation.separator.php
+          set:
+            - match: '(?={{path}})'
+              set:
+                - meta_scope: meta.path.php
+                - include: type-hint
+                - match: ''
+                  pop: 1
+        - match: '(?=\S)'
+          pop: 1
```

Comment:
Is there a special reason for this lookahead?

Can it be modified as follows?

```suggestion
    - match: ':'
      scope: punctuation.separator.php
      push:
        - match: (?={{path}})
          set:
            - meta_scope: meta.path.php
            - include: type-hint
            - match: ''
              pop: 1
```

Reply:
It is not needed at all, IMHO. The context is pushed only if a colon is matched. So the pattern in line 607 matches for sure and `set`s another context. So `(?=\S)` is never reached.

By removing the lookahead and `push`ing the path-context, nothing should change, if I don't miss a detail.

Reply:
I was thinking the same

Reply:
But what will happen if `(?={{path}})` doesn't match? Wouldn't it just kind of consume all rest of chars in the file?
```
    - match: ':'
      scope: punctuation.separator.php
      push:
        - match: (?={{path}})
          set:
            - meta_scope: meta.path.php
            - include: type-hint
            - match: ''
              pop: 1
```

Reply:
Nothing else, compared to current state of the PR, because lint 616 is not indented correctly to be able to pop in case of (?={{path}}) doesn't match.

Reply:
Agree, we should add it as alternative.

Reply:
```suggestion
    - match: ':'
      scope: punctuation.separator.php
      push:
        - match: (?={{path}})
          set:
            - meta_scope: meta.path.php
            - include: type-hint
            - match: ''
              pop: 1
        - match: (?=\S)
          pop: 1              
```

Reply:
Thanks!